### DISCUSSION TITLE
Is modifying `config.mdx_configs` allowed?

### DISCUSSION THREAD

**Discussion by rmorshea** [mkdocs/mkdocs#3226] [Q&A] [answered]
I've noticed some other discussions in which its suggested that, in order to view/modify markdown extension options you should use [`mdx_configs`](https://github.com/mkdocs/mkdocs/discussions/3196#discussioncomment-5703011). However, this attribute is marked as private and disallows assignment. Is it ok to mutate markdown extension configs? If so, should the docstring for this attribute be updated to explain this?

**Comment by kamilkrzyskow** [comment]
Private use means you shouldn't define it in the `mkdocs.yml` file. Try it and you'll get an error.
I think the Private type has some kind of "used internally by MkDocs" remark. Notice that you're already inside the internals of MkDocs, since you're accessing the `config` object the markdown extensions have already been split into their names and configs lists/dicts.

**Comment by oprypin** [comment, marked as answer]
Yes, you can modify `mdx_configs` in `on_config`. This will continue to be supported.

E.g. https://github.com/pawamoy/markdown-exec/blob/f7143854d7ec55d63dd55009a74b3ff275265628/src/markdown_exec/mkdocs_plugin.py#L69

**Reply by rmorshea** [reply]
I think [this docstring](https://github.com/mkdocs/mkdocs/blob/1f1bfad76336573ec3811ddba943ca0e9cc4cccf/mkdocs/config/defaults.py#L101) could use some improvement in that case. It seems to imply that `mdx_configs` should be treated as if it's private.

**Reply by oprypin** [reply]
@rmorshea 
* https://github.com/mkdocs/mkdocs/pull/3229
